simulation/halsim_ws_core/src/main/native/include/HALSimBaseWebSocketConnection.h

```diff
@@ -0,0 +1,29 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2020 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+#pragma once
+
+#include <memory>
+
+#include <wpi/json.h>
+
+namespace wpilibws {
+
+class HALSimBaseWebSocketConnection
+    : public std::enable_shared_from_this<HALSimBaseWebSocketConnection> {
+ public:
+  virtual void OnSimValueChanged(const wpi::json& msg) = 0;
+
+  std::shared_ptr<HALSimBaseWebSocketConnection> GetSharedFromThis() {
+    return shared_from_this();
+  }
```

Comment:
This function shouldn't be required, you can just call `shared_from_this()` directly?

Reply:
Yeah, I think I added it because there was `HALSimHttpConnection` derives from `HALSimBaseWebSocketConnection` and `wpi::HttpWebSocketServerConnection<derived>`, both of which derive from `enable_shared_from_this`. Not a huge problem since I can just invoke `HALSimBaseWebSocketConnection::shared_from_this()` in the server case